05-dom-document-loading

okonoshi opened the pull request:
Вроде что то вышло

okonoshi committed:
moved path definition to parent function

okonoshi committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20230223_aleksandr-velmozhko

okonoshi committed:
ColumnChart component

okonoshi committed:
re-writing

okonoshi committed:
Merge branch 'master' of https://github.com/js-tasks-ru/js-20230223_aleksandr-velmozhko

okonoshi committed:
completed  component NotificationMessage

okonoshi committed:
completed component NotificationMessage

okonoshi committed:
Merge branch 'master' of https://github.com/okonoshi/js-20230223_aleksandr-velmozhko

okonoshi committed:
first version of component implementation

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

okonoshi committed:
update 2-sortable-table-v1 version 1.0.1. 6/7 tests

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

okonoshi committed:
update 2-sortable-table-v1 version 1.0.2  7/7 tests

jsru commented:
Добавляю преподавателя (@dosandk) для код-ревью.

okonoshi committed:
update 2-sortable-table-v1 version 1.0.3 7/7 tests
added destructuring

jsru commented:
Решение было обновлено, посмотрим что скажет @dosandk

okonoshi committed:
deleted console.log

jsru commented:
Решение было обновлено, посмотрим что скажет @dosandk

dosandk commented on 04-oop-basic-intro-to-dom/1-column-chart/index.js:
```diff
@@ -1,99 +1,87 @@
+
+  getTemplate() {
+    const template = `
+    <div class="column-chart column-chart_loading"
```
что-то пошло не так с форматированием!

dosandk commented on 05-dom-document-loading/1-notification/index.js:
```diff
@@ -1,3 +1,53 @@
+  }
+
+  show(parentElement = document.body) {
+    let notification = document.querySelector(".notification");
```
почему `let`?

okonoshi replied:
Исправил

dosandk commented on 05-dom-document-loading/1-notification/index.js:
```diff
@@ -1,3 +1,53 @@
+  }
+
+  show(parentElement = document.body) {
+    let notification = document.querySelector(".notification");
```
🔴 please don't do this
так делать нельзя!
подумайте как проверить показ нотификейшена без обращения к `document`

okonoshi replied:
Исправил.
Но не понимаю почему тест не проходится без  show(parentElement = document.body), почему именно так нужно определять таргет элемент? А не через конструктор 
```
constructor(message = "", { duration = 0, type = "", targetElement = document.body} = {}) {
    this.targetElement = targetElement
  }
```

dosandk commented on 05-dom-document-loading/2-sortable-table-v1/index.js:
```diff
@@ -1,6 +1,166 @@
+      desc: -1,
+    };
+    this.updateArrow(field, order);
+    let compare;
```
может вынести в отдельный метод:
`getCompareFn`?

dosandk reviewed (changes requested):
решение `NotificationMessage` следует поправить,
посмотрите коммент

okonoshi committed:
Внес правки
Добавил статическое свойство в NotificationMessage
Вынес определение функции compare в отдельный метод в SortableTable

jsru commented:
Добавляю преподавателя (@dosandk) для код-ревью.

dosandk commented on 05-dom-document-loading/1-notification/index.js:
```diff
@@ -29,13 +30,13 @@ export default class NotificationMessage {
-    let notification = document.querySelector(".notification");
-    if (notification) {
-      notification.remove();
+  show(targetElement = document.body) {
```
можно и через конструктор реализовать, но в данном задание требование сделать это именно через метод `show`

dosandk reviewed (approved):
good 👍
